src/main.cpp

```diff
@@ -66,20 +147,19 @@ void SetRunOnStartup()
         return;
     }
 
-    PCWSTR keyname = L"Software\\Microsoft\\Windows\\CurrentVersion\\Run";
-    PCWSTR keyval = L"mykeepalive";
-
     if (runonstartup)
     {
         // Set the on startup path
         HKEY hkey = NULL;
-        RegCreateKey(HKEY_CURRENT_USER, keyname, &hkey);
-        RegSetValueEx(hkey, keyval, 0, REG_SZ, (BYTE*)(PWSTR)buffer, (DWORD)(wcslen((PWSTR)buffer) + 1) * 2);
+        if (RegCreateKey(HKEY_CURRENT_USER, Reg_KeyName, &hkey) == ERROR_SUCCESS) {
+            RegSetValueEx(hkey, Reg_KeyValue, 0, REG_SZ, (BYTE*)(PWSTR)buffer, (DWORD)(wcslen((PWSTR)buffer) + 1) * 2);
+            RegCloseKey(hkey);
```

Comment:
Note: You should call `RegCloseKey` on registry keys opened with `RegCreateKey`(`Ex`) or `RegOpenKey`(`Ex`).